ci/README.md

```diff
@@ -0,0 +1,202 @@
+
+### CirrusCI
+
+**Config**: [.cirrus.yml](https://github.com/dlang/dmd/blob/master/.cirrus.yml)
+
+**Checks**:
+- FreeBSD 11.4 x64, DMD (bootstrap)
+- FreeBSD 12.2 x64, DMD (coverage)
+- FreeBSD 12.2 x64, DMD (latest)
+- Ubuntu 18.04 x64, DMD (bootstrap)
+- Ubuntu 18.04 x64, DMD (latest)
+- Ubuntu 18.04 x64, GDC
+- Ubuntu 18.04 x64, LDC
+- Ubuntu 18.04 x86, DMD (bootstrap)
+- Ubuntu 18.04 x86, DMD (coverage)
+- Ubuntu 18.04 x86, DMD (latest)
+- macOS 10.15 x64, DMD (bootstrap)
+- macOS 11.x x64, DMD (coverage)
+- macOS 11.x x64, DMD (latest)
+- macOS 12.x x64, DMD (coverage)
+- macOS 12.x x64, DMD (latest)
+
+Cirrus tests DMD on Posix platforms.
+
+Since DMD is written in D, a "Host D Compiler" is needed to compile it.
+Various host compilers are tested, such as GDC, LDC, the latest DMD, and an older verison of DMD (bootstrap).
+
+Sometimes the macOS VMs get corrupted.
+When that happens, you can file an issue on Cirrus' issue tracker, [like this one](https://github.com/cirruslabs/macos-image-templates/issues/43).
+
+### pre-commit
+
+**Config**: [.pre-commit-config.yaml](https://github.com/dlang/dmd/blob/master/.pre-commit-config.yaml)
+
+**Checks**:
+- pre_commit
+
+pre-commit checks for redundant white space, such as trailing whitespace and newlines.
+
+If you want to automatically make your changes comply with the check, you can download `pre-commit` via your package manager, or with  `pip`.
+See https://pre-commit.com/.
+
+Run `pre-commit install` inside the root of the repository,
+This will add a hook into `.git/hooks/pre-commit`.
+Every time you do a commit it will do pre-commit checks.
+
+If you already have a custom `pre-commit` hook, you can modify the bash script to run `pre-commit` manually with `pre-commit run`.
+pre-commit is designed to run only on diffed files so it will run pretty quickly, but if you want to for it to run on all files you can do `pre-commit run --all-files`.
+
+### buildkite
+
+**Config:** [ci/buildkite](https://github.com/dlang/ci/tree/master/buildkite)
+
+**Checks**:
+- buildkite
+
+Buildkite tests that the compiler is still able to compile and pass the test suite of several popular D projects.
+
+The configuration file is found in a separate repository, since it is shared by druntime and Phobos.
+
+### CircleCI
+
+**Config**: [.circleci/config.yml](https://github.com/dlang/dmd/blob/master/.circleci/config.yml)
+
+**Checks**:
+- ci/circleci: build
+
+CircleCI tests DMD on Ubuntu 18.04.
+
+It also tests that the automatically generated C++ header, frontend.h, is updated.
+This is important because other compilers sharing the front-end (LDC and GDC) rely on DMD's header files to interface with it.
+When a PR modifies an `extern(C++)` function, the corresponding signature in a .h file should be updated as well, see [cxx-headers-test](https://github.com/dkorpel/dmd/tree/master/src#cxx-headers-test).
```

Comment:
This is a link to your fork not the main dmd repo.

Reply:
Fixed